### ISSUE TITLE
Feature suggestion : Loading button

### ISSUE THREAD

**Issue by storm1er** [Dogfalo/materialize#4941]
# TL;DR
Have `Component:loading-btn` which display a loading animation in it, +transition with existing `btn` stuff.
_Think it's a good idea ? add :+1:
Thanks ;)_

---

## Expected Behavior
I would like to see this kind of button :
![image](https://user-images.githubusercontent.com/9821946/28134425-375d18ce-6743-11e7-84b0-e8b7975da2b4.png)
See https://github.com/Dogfalo/materialize/pull/4942 for more

## UseCase
- In a form
- You click a *normal* button
- By adding class `is-loading` to it, it displays a loading animation
- Plus disabling button action, avoiding multi-submit & stuff
- Do your async/ajax/WTFYouWant
- Delete `is-loading` class and *YaY it's done*

## Material compliance
- Well, nothing is clearly stated about button & their motion
- But it's stay compliant if we respect [these motion rules](https://material.io/guidelines/motion/material-motion.html)

## Current Behavior
- Well there is no loading `btn` for the moment
- We could edit the content of a button with [mini preloader](http://materializecss.com/preloader.html) but it's not really a good solution as you need to edit DOM and too much stuff on html/css part of things.

## Possible Solution
Testable in [this fork](https://github.com/storm1er/materialize/tree/loading-btn)

**Comment by storm1er** [comment]
Just tried to edit css, and it's very doable.
Either right/left/none : 

![image](https://user-images.githubusercontent.com/9821946/27948094-b9925b28-62f8-11e7-941c-79b6a2fee588.png)

If people are interested in, i'll do it :)

**Comment by Dogfalo** [comment]
I would say initially that this seems a bit unnecessary but if we see people continuously ask for this, we'll consider it.

Will reopen this for now for visibility

**Comment by storm1er** [comment]
@Dogfalo ok :)
I actually need this in my project.
Too much async stuff, and nothing in materialize can handle those scenarios without too much dev.
So I'll work on it anyway ;)

**Comment by bert-w** [comment]
https://jsfiddle.net/tbsgvn6z/

```
<button class="red btn waves-effect">
  <div class="preloader-wrapper small active">
    <div class="spinner-layer spinner-blue-only">
      <div class="circle-clipper left">
        <div class="circle"></div>
      </div>
      <div class="gap-patch">
        <div class="circle"></div>
      </div>
      <div class="circle-clipper right">
        <div class="circle"></div>
      </div>
    </div>
  </div>
</button>
```

I know its not perfect, but with some css additions this can probably be fixed without re-creating existing components. `spinner-white-only` would be handy though

**Comment by storm1er** [comment]
@bert-w that's precisely what i want to avoid, DOM manipulate just for a loading animation feel really heavy.
BTW, i already told this in **current behavior** ^^'

**Comment by bassie1995** [comment]
I think this could be a pretty nice addition to make something that fits within Materialize well, but am currently using Ladda: https://github.com/hakimel/Ladda.

You can use the non-styled buttons and apply Materialize classes to make them look like Materialize buttons, but with the loading indicators.

**Comment by storm1er** [comment]
@bassie1995 Oh ^^
I'm just wondering why not implement Ladda in our framework ?
All those loading button are great and use-able easily.

Contract-overlay button is neat for submitting form without manually disabling every input & select.

Or just implement overlay capabilities on my fork ...

**Comment by Zerebokep** [comment]
Materialize definitely needs a loading button, I always wondered why there isn't one.

**Comment by Dogfalo** [comment]
See my comment over at #4942

**Comment by jpic** [comment]
Thanks a heap for your advice @Dogfalo keep up the great work !

**Comment by michelvd** [comment]
> I would say initially that this seems a bit unnecessary but if we see people continuously ask for this, we'll consider it.

I've never been on a project the last 10 years which didn't have a spinner-on-a-button. It is done on every action which might take some time ie a call to the server. Must have IMO
